lib/modules/ftp/weak_password/__init__.py

```diff
@@ -2,23 +2,50 @@
 # -*- coding: utf-8 -*-
 
 import time
-from core.compatible import generate_token
+import os
+import threading
+from core.exit_helper import terminate_thread
+from core.file_monitor import fileMonitor
 
 
 class ModuleProcessor:
     """
     this is the processor to run after docker machine is up to grab the log files or do other needed process...
     """
+
     def __init__(self):
         self.kill_flag = False
+        self.log_filename = 'tmp/ohp_ftp_weak_password_logs.txt'
+        self.log_filename_dump = 'tmp/ohp_ftp_weak_password_files_logs.json'
+        self.stop_execution = False
+        self.DIRECTORY_TO_WATCH = os.getcwd() + "/tmp/ohp_ftp_weak_container/"
+        self.EXCLUDES = ['/dev']
+        self.module_name = "ftp/weak_password"
+        if not os.path.exists(self.DIRECTORY_TO_WATCH):
+            os.makedirs(self.DIRECTORY_TO_WATCH)
 
     def processor(self):
         """
         processor function will be called as a new thread and will be die when kill_flag is True
         :return:
         """
+        newFileHandler = fileMonitor()
+        newFileHandler.log_filename = self.log_filename
+        newFileHandler.log_filename_dump = self.log_filename_dump
+        newFileHandler.DIRECTORY_TO_WATCH = self.DIRECTORY_TO_WATCH
+        newFileHandler.EXCLUDES = self.EXCLUDES
+        newFileHandler.module_name = self.module_name
```

Comment:
I noticed that these class variables in the ModuleProcessor class are being repeated, and they are there in the FileMonitor class well. There is an extra copy operation for all of them too, copying them from one to the other- l**og_filename, log_filename_dump,  DIRECTORY_TO_WATCH, EXCLUDES, module_name**. And then out of these, **log_filename, EXCLUDES** and **module_name** are further repeated in the ContainerFilesHandler and are copied once again when "_run_" function is called.

Any particular reason you have not created the object of the classes directly as a class variable? 

According to me, it might be more efficient to do it that way, instead of repeating and copying variable data.
An example would be-

```
class FileMonitor:
    def __init__(self):
        self.observer = Observer()
        self.log_filename_dump = None
        self.stop_execution = False
        self.DIRECTORY_TO_WATCH = None
        self.event_handler = containerFilesHandler()
```

```
class ModuleProcessor:
    def __init__(self):
        self.kill_flag = False
        self.stop_execution = False
        self.file_handler = FileMonitor()
        # And then set the values for the members.
        if not os.path.exists(self.file_handler.DIRECTORY_TO_WATCH):
            os.makedirs(self.file_handler.DIRECTORY_TO_WATCH)
```

And then directly set the values for - _self.event_handler.log_filename_ and other variables.

Reply:
Also, the naming convention for the variables should be kept consistent. "newFileHandler" should be "new_file_handler".

Reply:
@ChakshuGupta : i understand the repitition of the variables but it's kept purposely so that each class could be reused and if we pass variables from filemonitor to ContainerFilesHandler then it would be kind of lock-in.
also it cannot be created in init because event_handler should be defined in run I think because all watchdog examples show it this way and also it doesn't works fine when I tried to follow your initialization.

Reply:
Okay. As long as there is a reason behind it!